torchqdynamics/mesolve/mesolve.py

```diff
@@ -82,9 +82,17 @@ def mesolve(
     """
     # TODO H is assumed to be time-independent from here (temporary)
 
-    # convert H to a tensor and batch by default
-    H = to_tensor(H)
-    H_batched = H[None, ...] if H.dim() == 2 else H
+    if isinstance(H, OperatorLike.__args__):
+        # convert H to a tensor and batch by default
+        H = to_tensor(H)
+        H_batched = H[None, ...] if H.ndim == 2 else H
+    elif callable(H) or (isinstance(H, list) and (len(H) == 0 or callable(H[0]))):
+        # H is callable or list of callables
```

Comment:
If you want to add the bones for PWC H we need to edit the `TDOperatorLike` type.

Reply:
I suggest removing the part handling the piecewise constant `H~, because there are a few other things to change, and I would prefer that we do everything in the same PR, after this one.